### DISCUSSION TITLE
Build the Visual Studio project file

### DISCUSSION THREAD

**Discussion by lzh-maker** [OpenRadioss/OpenRadioss#1215] [Q&A]
Hello everyone,
I want to debug the engine project using Visual Studio under Windows, but I don't successfully generate the project file. I want to ask you that I used two ways to generate the project file, the first is to use build_script.sh, the input parameter is -arch=win64, but this method does not generate VS project file, but generates a makefile file; The second way is to use cmakegui and choose to generate Vs2019 project files, but the error ERROR is reported during the cmake process: -mpi= not available for this platform, do I have to install mpi to build? I'm confused about generating vs project files, is there a small partner who has successfully generated, can you share the experience?

**Comment by lzh-maker** [comment]
![3CF40DBA-D9D0-4ED1-936E-7E627A9FC3F0](https://user-images.githubusercontent.com/76392842/224626731-74cf398e-a84e-4259-94b1-c9d0e68efbc7.png)
![0C559DDB-ED65-4171-8729-C5F3D71F6BC9](https://user-images.githubusercontent.com/76392842/224626748-26fe0f18-6b74-4f72-b142-6d365dca4bdb.png)
I rebuilt it with cygwin, and the problem in this image appears, is this because of the low oneAPI version?

**Comment by OlivierWienholtz** [comment]
Hello,

I'll go through your questions : 

We encountered several issues with OneAPI versions prior to 2023.0.0.
I recommend to use OneAPI 2023.0.0 which is the latest available version...

but it should have compiled. 
You can add '-verbose' to build_script in order to dump verbose commands to the Makefiles.
I suspect Cygwin/Cmake  to have troubles with 'special characters' like th '@' in your path.

**Comment by OlivierWienholtz** [comment]
The build method with offer is dedicated to Unix & Co Tools (cygwin for windows native builds).
It is true that we don't offer native VS vcproj or MS-Build yet. 

The build with cmakegui is an option, pls add : mpi=none in the cmake configs.
IMO : there are several tasks to perform to have a native Windows build.
The build method implemented in Cmake is using Perl to generate some includes.
I doubt this is natively available on Intel Compiler / VS...
Those needs to to be rewritten/Ported to something more generic.

**Reply by lzh-maker** [reply]
![C4F12158-3974-4BC0-8941-2AAA800E919F](https://user-images.githubusercontent.com/76392842/224672605-f9faceb3-0d75-486b-9581-84895f41b6ee.png)
![9E039564-CFFF-4F59-A2F3-B785F0A132F4](https://user-images.githubusercontent.com/76392842/224672621-b6f745f9-9055-4ef9-bf04-ad4a56d2cc9e.png)
![6E3EF0DD-2863-4CB0-A81D-CF76CA9D3300](https://user-images.githubusercontent.com/76392842/224672628-2228168c-e1ce-4fa5-a6d4-49d75c6662f5.png)
I uninstalled the previous oneAPI and reinstalled a 2023.0 version, using "build_script.sh" to compile, plus the parameters "-verbose and -debug=1", and I still get the following error, I have mingw-win64 on my computer, I have a little question, will gfortran in mingw affect compilation? Will it conflict with gfortran in oneAPI?

**Reply by lzh-maker** [reply]
The symbol @ in the path has also been removed

**Reply by lzh-maker** [reply]
I also have a puzzle point, isn't the build result of the "build_script.sh" tool using -arch=win64 under windows in VS project file .sln format? My actual goal is to generate VS2019 project files, so that I can compile and debug under Windows more easily, so that I can better understand the code, can you tell me how to achieve my goal

**Reply by OlivierWienholtz** [reply]
Hello,

There is an issue in the script when 2 compilers are set & running eg : Intel OneAPI + MinGW.
I am looking to fix this.
For the time being I recommend to remove the PATH to cygwin, on Order Cygwin sees only Intel OneAPI.

I'll keep you informed.

On the other topic, I am currently checking what needs to be done to create svn or MSBuild files from Cmake but there might be some changes in our build scripts & methods to achieve this.
